File: api/src/main/java/net/impactdev/gts/api/util/WaitingInteger.java
package net.impactdev.gts.api.util;

public class WaitingInteger {

    private String fallback;
    private Integer result;

    public WaitingInteger(String fallback) {
        this.fallback = fallback;
    }

    public WaitingInteger(int result) {
        this.result = result;
    }

    @Override
    public String toString() {
        if(this.result != null) {
            return this.result.toString();
        }

        return this.fallback;
    }

}

File: api/src/main/kotlin/net/impactdev/gts/api/util/WaitingInteger.kt
package net.impactdev.gts.api.util

class WaitingInteger {
    private var fallback: String? = null
    private var result: Int? = null

    constructor(fallback: String?) {
        this.fallback = fallback
    }

    constructor(result: Int) {
        this.result = result
    }

    override fun toString(): String {
        return if (result != null) {
            result.toString()
        } else fallback!!
    }
}